precode_test.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func getResponse(url string) *httptest.ResponseRecorder {
+	req := httptest.NewRequest(http.MethodGet, url, nil)
+	resp := httptest.NewRecorder()
+	handler := http.HandlerFunc(mainHandle)
+	handler.ServeHTTP(resp, req)
+
+	return resp
+}
+
+func TestMainHandlerOk(t *testing.T) {
+	response := getResponse("/cafe?city=moscow&count=3")
+	targetCode := http.StatusOK
+	targetCount := 3
+
+	assert.Equal(t, targetCode, response.Code)
+	require.NotEmpty(t, response.Body)
```

Comment:
Так как в твоем случае это канал ридер
```suggestion
	require.NotEmpty(t, response.Body.String())
```